torch/fx/traceback.py

```diff
@@ -15,12 +20,76 @@
     "format_stack",
     "set_current_meta",
     "get_current_meta",
+    "NodeSource",
+    "get_graph_provenance_json",
 ]
 
 current_meta: Dict[str, Any] = {}
 should_preserve_node_meta = False
 
 
+class NodeSource:
+    """
+    NodeSource is a data structure that contains the provenance information of a node.
+    If node `a` is created from node `b`, then `a.meta["from_node"]` may contain NodeSource(b).
+    """
+
+    node_name: str
+    target: str
+    graph_id: int
+    pass_name: str
+    action: str
```

Comment:
Is it possible to have an enum here for action?

Reply:
Fixed now!